### PR TITLE
NEXUS-7995: Bump HttpClient et al. (2.11.x)

### PR DESCRIPTION
**Pull request by cstamas**
Update HttpClient and HttpCore to:
* HttpClient 4.3.6, [release notes](https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12310360&version=12327489)
* HttpCore 4.3.3, [release notes](https://issues.apache.org/jira/secure/ReleaseNote.jspa?projectId=12310340&version=12326260)

Issue
https://issues.sonatype.org/browse/NEXUS-7995

CI
http://bamboo.s/browse/NX-OSSF526

### CODE REVIEW

File: buildsupport/httpclient/pom.xml [hunk 1]
```diff
@@ -33,7 +33,7 @@
       <dependency>
         <groupId>org.apache.httpcomponents</groupId>
         <artifactId>httpclient</artifactId>
-        <version>4.3.5</version>
+        <version>4.3.6</version>
```
Comment (kellyrob99) [inline buildsupport/httpclient/pom.xml]: extract a property for this?

**Comment by kellyrob99** [comment]
+1

File: buildsupport/httpclient/pom.xml (hunk 1 above)
Comment (jdillon) [inline buildsupport/httpclient/pom.xml]: why?  this is used in 2 places? If it grows to more perhaps, but we don't need properties to manage versions for all dependencies, especially not in these tiny buildsupport modules.

**Comment by cstamas** [comment]
http://bamboo.s/browse/NX-OSSF526-1 :+1:

**Pull request merged**